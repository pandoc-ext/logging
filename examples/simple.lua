local logging = require 'logging'

function Pandoc(pandoc)
    logging.temp('pandoc', pandoc)
end
