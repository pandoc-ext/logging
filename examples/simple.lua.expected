(#) pandoc Pandoc {
  blocks: Blocks {
    [1] Header {
      attr: Attr {
        attributes: AttributeList {}
        classes: List {}
        identifier: "pandoc-lua-logging-examples"
      }
      content: Inlines {
        [1] Str "Pandoc"
        [2] Space
        [3] Str "lua"
        [4] Space
        [5] Str "logging"
        [6] Space
        [7] Str "examples"
      }
      level: 1
    }
    [2] Para {
      content: Inlines {
        [1] Str "This"
        [2] Space
        [3] Str "directory"
        [4] Space
        [5] Str "contains"
        [6] Space
        [7] Str "example"
        [8] Space
        [9] Str "lua"
        [10] Space
        [11] Str "files"
        [12] Space
        [13] Str "(standalone"
        [14] Space
        [15] Str "and"
        [16] Space
        [17] Str "pandoc"
        [18] Space
        [19] Str "filters)"
        [20] Space
        [21] Str "and"
        [22] Space
        [23] Str "some"
        [24] Space
        [25] Str "sample"
        [26] Space
        [27] Str "markdown"
        [28] Space
        [29] Str "files."
      }
    }
    [3] Para {
      content: Inlines {
        [1] Str "These"
        [2] Space
        [3] Str "files"
        [4] Space
        [5] Str "were"
        [6] Space
        [7] Str "used"
        [8] Space
        [9] Str "when"
        [10] Space
        [11] Str "generating"
        [12] Space
        [13] Str "the"
        [14] Space
        [15] Str "examples"
        [16] Space
        [17] Str "in"
        [18] Space
        [19] Str "the"
        [20] Space
        [21] Str "top-level"
        [22] Space
        [23] Str "README"
        [24] Space
        [25] Str "file."
      }
    }
    [4] Para {
      content: Inlines {
        [1] Str "Thereâ\128\153s"
        [2] Space
        [3] Str "also"
        [4] Space
        [5] Str "a"
        [6] Space
        [7] Code {
          attr: Attr {
            attributes: AttributeList {}
            classes: List {}
            identifier: ""
          }
          text: "Makefile"
        }
        [8] Space
        [9] Str "that"
        [10] Space
        [11] Str "runs"
        [12] Space
        [13] Str "lua"
        [14] Space
        [15] Str "and"
        [16] Space
        [17] Str "pandoc"
        [18] Space
        [19] Str "(as"
        [20] Space
        [21] Str "appropriate)"
        [22] Space
        [23] Str "and"
        [24] Space
        [25] Str "compares"
        [26] Space
        [27] Str "the"
        [28] Space
        [29] Str "output"
        [30] Space
        [31] Str "with"
        [32] Space
        [33] Str "the"
        [34] Space
        [35] Str "expected"
        [36] Space
        [37] Str "output."
      }
    }
    [5] Header {
      attr: Attr {
        attributes: AttributeList {
          a: "b"
        }
        classes: List {
          [1] "new-page"
        }
        identifier: "header"
      }
      content: Inlines {
        [1] Str "Header"
      }
      level: 1
    }
    [6] Para {
      content: Inlines {
        [1] Str "Paragraph"
        [2] Space
        [3] Str "with"
        [4] Space
        [5] Emph {
          content: Inlines {
            [1] Str "emphasized"
          }
        }
        [6] Space
        [7] Str "and"
        [8] Space
        [9] Strong {
          content: Inlines {
            [1] Str "bold"
          }
        }
        [10] Space
        [11] Str "text."
      }
    }
    [7] CodeBlock {
      attr: Attr {
        attributes: AttributeList {}
        classes: List {}
        identifier: ""
      }
      text: "Code"
    }
    [8] BulletList {
      content: List {
        [1] Blocks {
          [1] Plain {
            content: Inlines {
              [1] Str "and"
              [2] Space
              [3] Str "a"
              [4] Space
              [5] Str "list."
            }
          }
        }
      }
    }
    [9] Para {
      content: Inlines {
        [1] Str "text"
      }
    }
    [10] Table {
      attr: Attr {
        attributes: AttributeList {}
        classes: List {}
        identifier: ""
      }
      bodies: List {
        [1] {
          attr: Attr {
            attributes: AttributeList {}
            classes: List {}
            identifier: ""
          }
          body: List {
            [1] pandoc.Row {
              attr: Attr {
                attributes: AttributeList {}
                classes: List {}
                identifier: ""
              }
              cells: List {
                [1] pandoc.Cell {
                  alignment: "AlignDefault"
                  attr: Attr {
                    attributes: AttributeList {}
                    classes: List {}
                    identifier: ""
                  }
                  col_span: 1
                  contents: Blocks {
                    [1] Plain {
                      content: Inlines {
                        [1] Str "Tom"
                      }
                    }
                  }
                  row_span: 1
                }
                [2] pandoc.Cell {
                  alignment: "AlignDefault"
                  attr: Attr {
                    attributes: AttributeList {}
                    classes: List {}
                    identifier: ""
                  }
                  col_span: 1
                  contents: Blocks {
                    [1] Plain {
                      content: Inlines {
                        [1] Str "Black"
                        [2] Space
                        [3] Str "and"
                        [4] Space
                        [5] Str "white"
                      }
                    }
                  }
                  row_span: 1
                }
              }
            }
          }
          head: List {}
          row_head_columns: 0
        }
      }
      caption: {
        long: Blocks {
          [1] Plain {
            content: Inlines {
              [1] Str "Cats"
            }
          }
        }
      }
      colspecs: List {
        [1] {
          [1] "AlignDefault"
        }
        [2] {
          [1] "AlignDefault"
        }
      }
      foot: pandoc.TableFoot {
        attr: Attr {
          attributes: AttributeList {}
          classes: List {}
          identifier: ""
        }
        rows: List {}
      }
      head: pandoc.TableHead {
        attr: Attr {
          attributes: AttributeList {}
          classes: List {}
          identifier: ""
        }
        rows: List {
          [1] pandoc.Row {
            attr: Attr {
              attributes: AttributeList {}
              classes: List {}
              identifier: ""
            }
            cells: List {
              [1] pandoc.Cell {
                alignment: "AlignDefault"
                attr: Attr {
                  attributes: AttributeList {}
                  classes: List {}
                  identifier: ""
                }
                col_span: 1
                contents: Blocks {
                  [1] Plain {
                    content: Inlines {
                      [1] Str "Name"
                    }
                  }
                }
                row_span: 1
              }
              [2] pandoc.Cell {
                alignment: "AlignDefault"
                attr: Attr {
                  attributes: AttributeList {}
                  classes: List {}
                  identifier: ""
                }
                col_span: 1
                contents: Blocks {
                  [1] Plain {
                    content: Inlines {
                      [1] Str "Description"
                    }
                  }
                }
                row_span: 1
              }
            }
          }
        }
      }
    }
  }
  meta: Meta {
    foo: Inlines {
      [1] Emph {
        content: Inlines {
          [1] Str "emph"
        }
      }
    }
    goo: Inlines {
      [1] Underline {
        content: Inlines {
          [1] Str "underline"
        }
      }
    }
  }
}
